Update istioingress  config howto for external listeners due to istio operator v2 (#55)
banzaicloud/koperator-docs: docs/external-listener/index.md

          accessMethod: LoadBalancer
    ```

1. Set the ingress controller. The ingress controllers that are currently supported for load balancing are:

    - `envoy`: uses Envoy proxy as an ingress.

>     - `istioingress`: uses Istio proxy gateway as an ingress. Istio ingress is the default controller for Kafka clusters provisioned with [Supertubes](/docs/supertubes/overview/), since those clusters run inside an Istio mesh.

Reviewer: ```suggestion
    - `istioingress`: uses Istio proxy gateway as an ingress. Istio ingress is the default controller for Kafka clusters provisioned with [SDM](/docs/supertubes/overview/), since those clusters run inside an Istio mesh.
```

    Configure the ingress controller you want to use:

>     - To use Envoy, set the `ingressController` field in the `KafkaCluster` custom resource to `envoy`. For an example, see [https://github.com/banzaicloud/koperator/blob/master/config/samples/banzaicloud_v1beta1_kafkacluster.yaml](https://github.com/banzaicloud/koperator/blob/master/config/samples/banzaicloud_v1beta1_kafkacluster.yaml)

Reviewer: ```suggestion
    - To use Envoy, set the `ingressController` field in the `KafkaCluster` custom resource to `envoy`. For an example, [see](https://github.com/banzaicloud/koperator/blob/master/config/samples/banzaicloud_v1beta1_kafkacluster.yaml)
```

Reviewer: Instead of pointing to an example yaml I think we should create some yaml snippet here with the required fields. To follow the pattern already inside the documentation. At the end maybe we can point the user to an example.

>     - To use Istio ingress controller set the `ingressController` field to `istioingress`. Istio operator v2 is supported from Koperator version 0.21.0+. Because of this, you must also configure the `istioControlPlane` field in the `KafkaCluster` custom resource. `istioControlPlane` is a reference to the IstioControlPlane resource. For an example, see [https://github.com/banzaicloud/koperator/blob/master/config/samples/kafkacluster-with-istio.yaml](https://github.com/banzaicloud/koperator/blob/master/config/samples/kafkacluster-with-istio.yaml)

Reviewer: ```suggestion
    - To use Istio ingress controller set the `ingressController` field to `istioingress`. Istio operator v2 is supported from Koperator version 0.21.0+. Because of this, you must also configure the `istioControlPlane` field in the `KafkaCluster` custom resource. `istioControlPlane` is a reference to the IstioControlPlane resource. For an example, [see](https://github.com/banzaicloud/koperator/blob/master/config/samples/kafkacluster-with-istio.yaml)
```

>     - To use Istio ingress controller set the `ingressController` field to `istioingress`. Istio operator v2 is supported from Koperator version 0.21.0+. Because of this, you must also configure the `istioControlPlane` field in the `KafkaCluster` custom resource. `istioControlPlane` is a reference to the IstioControlPlane resource. For an example, see [https://github.com/banzaicloud/koperator/blob/master/config/samples/kafkacluster-with-istio.yaml](https://github.com/banzaicloud/koperator/blob/master/config/samples/kafkacluster-with-istio.yaml)

Reviewer: I would definitely mention or maybe even link that what is istio operator v2.

Reviewer: Also can we maybe elaborate more on what this field(istioControlPlane) does and why it is required?

          accessMethod: LoadBalancer
    ```

1. Set the ingress controller. The ingress controllers that are currently supported for load balancing are:

    - `envoy`: uses Envoy proxy as an ingress.
    - `istioingress`: uses Istio proxy gateway as an ingress. Istio ingress is the default controller for Kafka clusters provisioned with [SDM](/docs/supertubes/overview/), since those clusters run inside an Istio mesh.

    Configure the ingress controller you want to use:

    - To use Envoy, set the `ingressController` field in the `KafkaCluster` custom resource to `envoy`. For an example, [see](https://github.com/banzaicloud/koperator/blob/master/config/samples/banzaicloud_v1beta1_kafkacluster.yaml).

      ```yaml
      spec:
        ingressController: "envoy"
      ```

>     - To use Istio ingress controller set the `ingressController` field to `istioingress`. [Istio operator](https://github.com/banzaicloud/istio-operator) v2 is supported from Koperator version 0.21.0+. Istio operator manage Istio resources to make available istio ingress. Istio operator v2 supports multiple Istio control plane on a same cluster, that is why the corresponding control plane to the gateway must be specified. The `istioControlPlane` field in the `KafkaCluster` custom resource is a reference to that IstioControlPlane resource. For an example, [see](https://github.com/banzaicloud/koperator/blob/master/config/samples/kafkacluster-with-istio.yaml).

Reviewer: ```suggestion
    - To use Istio ingress controller set the `ingressController` field to `istioingress`. [Istio operator](https://github.com/banzaicloud/istio-operator) v2 is supported from Koperator version 0.21.0+. Istio operator manage Istio resources to make available istio ingress. Istio operator v2 supports multiple Istio control plane on a same cluster, that is why the corresponding control plane to the gateway must be specified. The `istioControlPlane` field in the `KafkaCluster` custom resource is a reference to that IstioControlPlane resource. For an example, [see](https://github.com/banzaicloud/koperator/blob/672b19d49e5c0a22f9658181003beddb56f17d33/config/samples/kafkacluster-with-istio.yaml#L10).
```

>     - To use Istio ingress controller set the `ingressController` field to `istioingress`. [Istio operator](https://github.com/banzaicloud/istio-operator) v2 is supported from Koperator version 0.21.0+. Istio operator manage Istio resources to make available istio ingress. Istio operator v2 supports multiple Istio control plane on a same cluster, that is why the corresponding control plane to the gateway must be specified. The `istioControlPlane` field in the `KafkaCluster` custom resource is a reference to that IstioControlPlane resource. For an example, [see](https://github.com/banzaicloud/koperator/blob/master/config/samples/kafkacluster-with-istio.yaml).

Reviewer: ```suggestion
    - To use Istio ingress controller set the `ingressController` field to `istioingress`. [Istio operator](https://github.com/banzaicloud/istio-operator) v2 is supported from Koperator version 0.21.0+. Istio operator v2 supports multiple Istio control plane on the same cluster, that is why the corresponding control plane to the gateway must be specified. The `istioControlPlane` field in the `KafkaCluster` custom resource is a reference to that IstioControlPlane resource. For an example, [see](https://github.com/banzaicloud/koperator/blob/master/config/samples/kafkacluster-with-istio.yaml).
```

      ```yaml
      spec:
        ingressController: "istioingress"
        istioControlPlane:
          name: <name of the IstioControlPlane resource>

>           namespace: <namespace of the IstioControlPlane resource>

Reviewer 2: ```suggestion
          namespace: <namespace of the IstioControlPlane custom resource>
```
